lib/modem_key_mgmt/modem_key_mgmt.c

```diff
@@ -65,20 +65,31 @@ static int translate_error(int err)
 
 	/* In case of CME error translate to an errno value */
 	switch (nrf_modem_at_err(err)) {
-	case 513:
+	case 513: /* not found */
+		LOG_WRN("Key not found");
 		return -ENOENT;
-	case 514:
-		return -EPERM;
-	case 515:
-		return -ENOMEM;
-	case 518:
+	case 514: /* no access */
+		LOG_WRN("Key access refused");
 		return -EACCES;
+	case 515: /* memory full */
+		LOG_WRN("Key storage memory full");
+		return -ENOMEM;
+	case 518: /* not allowed in active state */
+		LOG_WRN("Not allowed when LTE connection is active");
+		return -EPERM;
+	case 519: /* already exists */
+		LOG_WRN("Key already exists");
+		return -EALREADY;
+	case 528: /* not allowed in power off warning */
+		LOG_WRN("Not allowed when power off warning is active");
+		return -ECANCELED;
 	default:
 		/* Catch unexpected CME errors.
 		 * Return a magic value to make sure this
 		 * situation is clearly distinguishable.
 		 */
-		__ASSERT(false, "Untranslated CME error %d!", nrf_modem_at_err(err));
+		LOG_ERR("Untranslated CME error %d", nrf_modem_at_err(err));
+		__ASSERT(false, "Untranslated CME error %d", nrf_modem_at_err(err));
```

Comment:
So are these unexpected error codes based on RPC spec or what?
Just trying to understand whether assert is too radical action here.

Reply:
No, they are possible CME values from the AT command [documentation](https://infocenter.nordicsemi.com/index.jsp?topic=%2Fref_at_commands%2FREF%2Fat_commands%2Fsecurity%2Fcmng_set.html).

Reply:
Ok, I guess this is fine as I expect we haven't hit this assert. Just thinking what happens if a modem revision brings a new return value and libmodem and modem version is a bit out of synch in the device.

Reply:
libmodem's out of the picture here. If a new modem introduces a new error, it depends first and foremost if the application has `__ASSERT` enabled. In which case it will assert, unless the application is also updated. If the asserts are disabled, then the application will see an error it can't make much sense of, then it's up to the application how that is handled.